Sources/Noora/Utilities/Terminal.swift

```diff
@@ -86,6 +90,25 @@ public struct Terminal: Terminaling {
         return char != EOF ? Character(UnicodeScalar(UInt8(char))) : nil
     }
 
+    public func readCharacterNonBlocking() -> Character? {
```

Comment:
I'd add some basic documentation for this method as it's not immediately obvious what non-blocking means in this context.